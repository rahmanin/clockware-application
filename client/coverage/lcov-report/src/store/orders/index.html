
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/store/orders</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/store/orders</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.64% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>39/44</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.24% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>37/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.ts"><a href="actions.ts.html">actions.ts</a></td>
	<td data-value="73.68" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.68" class="pct medium">73.68%</td>
	<td data-value="19" class="abs medium">14/19</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	</tr>

<tr>
	<td class="file high" data-value="reducer.ts"><a href="reducer.ts.html">reducer.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Dec 01 2020 12:22:24 GMT+0200 (Eastern European Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    