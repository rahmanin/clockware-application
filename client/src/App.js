import React, {useEffect} from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import Loader from "./components/Loader";
import Header from './components/Header';
import Content from "./components/Content";
import Masters from "./admin/Masters";
import Orders from "./admin/Orders";
import Cities from "./admin/Cities";
import Prices from "./admin/Prices";
import Diagrams from "./admin/Diagrams";
import { routes } from "./constants/routes";
import MakingOrder from "./routes/OrderRoute";
import LogIn from './routes/LogInRoute';
import Feedback from './routes/FeedbackRoute';
import ChooseMaster from "./routes/ChooseMasterRoute";
import jwtDecode from 'jwt-decode';
import {useDispatch} from "react-redux";
import { useSelector } from "react-redux";
import {checkToken} from "./store/users/actions";
import {userParams, userLoading} from "./store/users/selectors";

import './App.scss';

export default function App() {
  const dispatch = useDispatch();
  const userData = useSelector(userParams)
  const userIsLoading = useSelector(userLoading)

  useEffect(() => {
    localStorage.token && dispatch(checkToken())
  }, []);
  
  const checkAuth = () => {
    const token = localStorage.token;
    var isLogged = false;
    if (token) {
      const tokenExpiration = jwtDecode(token).exp;
      const dateNow = new Date();
      if (tokenExpiration < dateNow.getTime()/1000) {
        isLogged = false;
      } else {
        isLogged = true;
      }
    } else {
      isLogged = false;
    }
    return isLogged;
  }

  const {
    order, 
    chooseMaster, 
    login, 
    masters, 
    orders, 
    cities, 
    prices, 
    feedback, 
    diagrams
  } =  routes;
  
  if (userIsLoading) return <Loader />

  return (
    <Router>
      <Header />
      <Content>
        <Switch>
          <Redirect exact from="/" to={checkAuth() ? orders : order} />
          <Route path={order} exact component={MakingOrder}/>
          <Route path={chooseMaster} exact component={ChooseMaster}/>
          <Route path={feedback} exact component={Feedback}/>
          <Route path={login} exact component={LogIn}/>
          <Route path={masters} render={() => checkAuth() && userData && userData.is_admin ? (<Masters />) : (<Redirect to={orders}/>)}/>
          <Route path={orders} render={() => checkAuth() ? (<Orders />) : (<Redirect to={login}/>)}/>
          <Route path={cities} render={() => checkAuth() && userData && userData.is_admin ? (<Cities />) : (<Redirect to={orders}/>)}/>
          <Route path={diagrams} render={() => checkAuth() && userData && userData.is_admin ? (<Diagrams />) : (<Redirect to={orders}/>)}/>
          <Route path={prices} render={() => checkAuth() && userData && userData.is_admin ? (<Prices />) : (<Redirect to={orders}/>)}/>
        </Switch>
      </Content>
    </Router>
);
}