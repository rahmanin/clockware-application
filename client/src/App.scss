* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body { 
  background-position: center;
  background-size: cover;
  font-family: 'Roboto';
}
