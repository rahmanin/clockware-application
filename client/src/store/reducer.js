import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
});

export default rootReducer;