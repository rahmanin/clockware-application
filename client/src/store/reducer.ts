import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";
import usersReducer from "./users/reducer";
import citiesReducer from "./cities/reducer";
import mastersReducer from "./masters/reducer";
import ordersReducer from "./orders/reducer";
import ordersClientReducer from "./ordersClient/reducer";
import { createStore, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension'
import newsReducer from "./news/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
  user: usersReducer,
  cities: citiesReducer,
  masters: mastersReducer,
  orders: ordersReducer,
  ordersClient: ordersClientReducer,
  news: newsReducer
});

const composedEnhancer = composeWithDevTools(
  applyMiddleware(thunk)
)

const store = createStore(
  rootReducer,
  composedEnhancer
);

export default store
