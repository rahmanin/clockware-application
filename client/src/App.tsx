import React, {useEffect} from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import {Loader} from "./components/Loader";
import {Header} from './components/Header';
import {Content} from "./components/Content";
import {Masters} from "./admin/Masters";
import {Orders} from "./admin/Orders";
import {Cities} from "./admin/Cities";
import {Prices} from "./admin/Prices";
import {Diagrams} from "./admin/Diagrams";
import { routes } from "./constants/routes";
import MakingOrder from "./routes/OrderRoute";
import LogIn from './routes/LogInRoute';
import {Blog} from './routes/BlogRoute';
import Feedback from './routes/FeedbackRoute';
import {BlogEditor} from './admin/BlogEditorRoute';
import ChooseMaster from "./routes/ChooseMasterRoute";
import jwtDecode from 'jwt-decode';
import {useDispatch} from "react-redux";
import { useSelector } from "react-redux";
import {checkToken} from "./store/users/actions";
import {userParams, userLoading} from "./store/users/selectors";
import {UserData} from "./store/users/actions"
import './App.scss';

export default function App() {
  const dispatch = useDispatch();
  const userData: UserData = useSelector(userParams)
  const userIsLoading: boolean = useSelector(userLoading)
  const isAdmin: boolean = userData && userData.role === "admin"

  useEffect(() => {
    localStorage.token && dispatch(checkToken())
  }, []);
  
  const checkAuth = () => {
    const token: string = localStorage.token;
    var isLogged: boolean = false;
    if (token) {
      const tokenExpiration: number = jwtDecode<any>(token).exp;
      const dateNow: Date = new Date();
      if (tokenExpiration < dateNow.getTime()/1000) {
        isLogged = false;
      } else {
        isLogged = true;
      }
    } else {
      isLogged = false;
    }
    return isLogged;
  }

  const {
    order, 
    chooseMaster, 
    login, 
    masters, 
    orders, 
    cities, 
    prices, 
    feedback, 
    diagrams,
    blogEditor,
    blog
  } =  routes;
  
  if (userIsLoading) return <Loader />

  return (
    <Router>
      <Header />
      <Content>
        <Switch>
          <Redirect exact from="/" to={checkAuth() ? orders : order} />
          <Route path={order} exact component={MakingOrder}/>
          <Route path={chooseMaster} exact component={ChooseMaster}/>
          <Route path={feedback} exact component={Feedback}/>
          <Route path={login} exact component={LogIn}/>
          <Route path={blog} exact component={Blog}/>
          <Route path={blogEditor} render={() => checkAuth() && isAdmin ? (<BlogEditor />) : (<Redirect to={orders}/>)}/>
          <Route path={masters} render={() => checkAuth() && isAdmin ? (<Masters />) : (<Redirect to={orders}/>)}/>
          <Route path={orders} render={() => checkAuth() ? (<Orders />) : (<Redirect to={login}/>)}/>
          <Route path={cities} render={() => checkAuth() && isAdmin ? (<Cities />) : (<Redirect to={orders}/>)}/>
          <Route path={diagrams} render={() => checkAuth() && isAdmin ? (<Diagrams />) : (<Redirect to={orders}/>)}/>
          <Route path={prices} render={() => checkAuth() && isAdmin ? (<Prices />) : (<Redirect to={orders}/>)}/>
        </Switch>
      </Content>
    </Router>
  );
}