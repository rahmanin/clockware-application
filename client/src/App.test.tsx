import React from 'react';
import {cleanup} from '@testing-library/react';
import { Provider } from "react-redux";
import store from "./store/reducer";
import { act } from "react-dom/test-utils";
import {mount} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });
afterEach(cleanup);

act(() => {
  it('Render App', () => {
    const component = mount(
      <Provider store={store}>
        <App/>
      </Provider>
    );
    expect(component.debug()).toMatchSnapshot();
  });
})
