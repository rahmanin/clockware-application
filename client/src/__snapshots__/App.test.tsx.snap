// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
"<Provider store={{...}}>
  <App>
    <BrowserRouter>
      <Router history={{...}}>
        <Header>
          <div className=\\"header\\" id=\\"header\\">
            <Navbar>
              <styled.nav>
                <nav className=\\"sc-dlfnbm gnAerA\\">
                  <Link to=\\"/order\\">
                    <LinkAnchor href=\\"/order\\" navigate={[Function: navigate]}>
                      <a href=\\"/order\\" onClick={[Function: onClick]}>
                        <div className=\\"logo\\">
                          CLOCKWARE
                        </div>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <Burger>
                    <styled.div open={false} onClick={[Function: onClick]}>
                      <div open={false} onClick={[Function: onClick]} className=\\"sc-gsTCUz bvOIHq\\">
                        <div />
                        <div />
                        <div />
                      </div>
                    </styled.div>
                    <RightNav open={false} onClick={[Function: onClick]}>
                      <styled.ul open={false}>
                        <ul open={false} className=\\"sc-bdfBwQ jmMyGP\\">
                          <li hidden={true}>
                            <Link to=\\"/diagrams\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/diagrams\\" navigate={[Function: navigate]}>
                                <a href=\\"/diagrams\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Statistics
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li hidden={true}>
                            <Link to=\\"/masters\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/masters\\" navigate={[Function: navigate]}>
                                <a href=\\"/masters\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Masters
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li hidden={true}>
                            <Link to=\\"/cities\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/cities\\" navigate={[Function: navigate]}>
                                <a href=\\"/cities\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Cities
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li hidden={true}>
                            <Link to=\\"/orders\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/orders\\" navigate={[Function: navigate]}>
                                <a href=\\"/orders\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Orders
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li hidden={true}>
                            <Link to=\\"/prices\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/prices\\" navigate={[Function: navigate]}>
                                <a href=\\"/prices\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Prices
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li>
                            <Link to=\\"/blog\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/blog\\" navigate={[Function: navigate]}>
                                <a href=\\"/blog\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Blog
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li hidden={true}>
                            <Link to=\\"/blogEditor\\" onClick={[Function: onClick]}>
                              <LinkAnchor onClick={[Function: onClick]} href=\\"/blogEditor\\" navigate={[Function: navigate]}>
                                <a href=\\"/blogEditor\\" onClick={[Function: onClick]}>
                                  <div className=\\"links\\">
                                    Blog editor
                                  </div>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                          <li>
                            <div className=\\"links\\" onClick={[Function: logIn]}>
                              Log in
                            </div>
                          </li>
                        </ul>
                      </styled.ul>
                    </RightNav>
                  </Burger>
                </nav>
              </styled.nav>
            </Navbar>
          </div>
        </Header>
        <Content>
          <div className=\\"content\\">
            <Switch>
              <Route path=\\"/order\\" exact={true} component={[Function: MakingOrder]} location={{...}} computedMatch={{...}}>
                <MakingOrder history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                  <Loader>
                    <div className=\\"isLoading\\">
                      <div className=\\"circle c_1\\" />
                      <div className=\\"circle c_2\\" />
                      <div className=\\"circle c_3\\" />
                    </div>
                  </Loader>
                </MakingOrder>
              </Route>
            </Switch>
          </div>
        </Content>
      </Router>
    </BrowserRouter>
  </App>
</Provider>"
`;
